### ISSUE TITLE
Deterministic = True returns NaN trades

### ISSUE THREAD

**Issue by looguzp** [AI4Finance-Foundation/FinRL#453]
Hi, I was trying to make a simple one stock DRL experiment. I used Yahoo AAPL data and trained my code in following fasihon
```

`eval_callback = EvalCallback( VecMonitor(e_trade_gym_sb), best_model_save_path='./logs/',
                                 log_path='./logs/', eval_freq=2000,
                                 deterministic=True, render=True)
```

```

    model=A2C(CustomActorCriticPolicy,env=env_train,use_rms_prop=True,verbose=1,n_steps=25,learning_rate=0.0003,gae_lambda=0.98,gamma=0.95,ent_coef=0.01)
    model.learn(30000,callback=eval_callback)`
```
However whenever I load the saved model and predict using deterministic = True agent dont do any trade at all and the backtest returns Nan values.  Is there any solution to this issue. I check some other issues with the same problem but not yer answered. 

Regards

**Comment by Athe-kunal** [comment]
Can you provide a snippet of CustomActorPolicy? SB3 is compatible with MlpPolicy only, so I just wanted to check your custom policy

**Comment by looguzp** [comment]
hi sorry for late response here is the policy I used. Its a MLPPolicy that I took from the sb3 website and modified a little bit

`class CustomNetwork(nn.Module):
    """
    Custom network for policy and value function.
    It receives as input the features extracted by the feature extractor.

    :param feature_dim: dimension of the features extracted with the features_extractor (e.g. features from a CNN)
    :param last_layer_dim_pi: (int) number of units for the last layer of the policy network
    :param last_layer_dim_vf: (int) number of units for the last layer of the value network
    """

    def __init__(
        self,
        feature_dim: int,
        last_layer_dim_pi: int = 16,
        last_layer_dim_vf: int = 16,

        first_layer_dim_pi: int = 16,
        first_layer_dim_vf: int = 16,

        second_layer_dim_pi: int = 16,
        second_layer_dim_vf: int = 16,



    ):
        super(CustomNetwork, self).__init__()

        # IMPORTANT:
        # Save output dimensions, used to create the distributions
        self.latent_dim_pi = last_layer_dim_pi
        self.latent_dim_vf = last_layer_dim_vf

        # Policy network
        self.policy_net = nn.Sequential(
            nn.Linear(feature_dim, first_layer_dim_pi), nn.ReLU(),
            nn.Linear(first_layer_dim_pi, second_layer_dim_pi), nn.ReLU(),
            nn.Dropout(0.5),
            nn.Linear(second_layer_dim_pi, last_layer_dim_pi), nn.Tanh()
        )
        # Value network
        self.value_net = nn.Sequential(
            nn.Linear(feature_dim, first_layer_dim_vf), nn.ReLU(),
            nn.Linear(first_layer_dim_vf, second_layer_dim_vf), nn.ReLU(),
            nn.Dropout(0.5),
            nn.Linear(second_layer_dim_vf, last_layer_dim_vf), nn.Tanh()
        )

        self.value_net.apply(self.initialize_weights)
        self.policy_net.apply(self.initialize_weights)

    def forward(self, features: th.Tensor) -> Tuple[th.Tensor, th.Tensor]:
        """
        :return: (th.Tensor, th.Tensor) latent_policy, latent_value of the specified network.
            If all layers are shared, then ``latent_policy == latent_value``
        """
        return self.policy_net(features), self.value_net(features)

    def forward_actor(self, features: th.Tensor) -> th.Tensor:
        return self.policy_net(features)

    def forward_critic(self, features: th.Tensor) -> th.Tensor:
        return self.value_net(features)

    def initialize_weights(self,m):

        if isinstance(m, nn.Linear):
            nn.init.kaiming_uniform_(m.weight.data)
            nn.init.constant_(m.bias.data, 0)`

**Comment by Athe-kunal** [comment]
Either the model has diverged or there is some error in the training loop. Please go over it again
